### ISSUE TITLE
Will Ubuntu 18.04 LTS be available with the Spring Release?

### ISSUE THREAD

**Issue by DarthSpock** [Microsoft/WSL#2874]
Ubuntu officially releases 18.04 LTS in April. I was wondering if it will be published to the Windows Store when the Spring release is available and whether it will require the Spring release as a dependency or will it be available in FCU?

**Comment by Biswa96** [comment]
I think that MS programmers need to update LxCore.sys (i.e. Linux kernel) to support updated distribution and more syscalls.

**Comment by benhillis** [comment]
We will be looking at 18.04 support but do not have an official timeline yet.

**Comment by DarthSpock** [comment]
Perhaps resolving #1579 and getting a working `systemd` that can allow unblocking of issues such as #2533 could go a long way to supporting it. We can track support for 18.04 here (the more visibility on this, the better).

**Comment by DarthSpock** [comment]
An update to this since I discovered from Ubuntu's twitter feed some pertinent information. Apparently the default for Bionic Beaver shall be Xorg vice Wayland. An interesting development and a regression in my opinion. So supporting 18.04 will be easier GUI wise if MS is doing any work supporting X. I would still advocate Wayland support though. It may not be default but it will still be available unless Canonical is deciding to drop it from their repositories (very unlikely).

**Comment by therealkenc** [comment]
Xorg/Wayland has zero, nada, nothing, bupkis, to do with whether or when Bionic makes it to the Windows Store. And not because "WSL doen't support drawing pixels". One just literally has nothing to do with the other. Twenty years from now you'll still be able to type `xeyes` into your terminal and it will still pop up. Nothing can make `libX11` stop working because it is a socket library (anymore than `libcurl` will stop working). Likewise `libgtk*` will fall back to X protocol for the foreseeable (and likely even the unforeseeable) future. It just doesn't matter.

**Comment by DarthSpock** [comment]
Agreed, they are unrelated. I just put the info there as it pertained to the upcoming LTS release. With the launcher now open-source, it may not be as relevant if it's available in the Store now though of course it would still be nice if they updated their code to install 18.04 vice 16.04 and thus save time and effort for everyone. Or we can just `do-release-upgrade` since the clone issues were fixed.

**Comment by iBug** [comment]
Try this:

```
sudo do-release-upgrade
```

If it says no new version found, supply `-d` as an argument.

I just upgraded my WSL to Ubuntu 18.04 and enjoying it.

**Comment by MikeGitb** [comment]
Any new official information on this? it be particularly useful, if we could install different versions side by side.

**Comment by 9527KKK** [comment]
I don't like "sudo do-release-upgrade" since it does not remove old/deprecated software.. I just want a clean Ubunutu 18.04 WSL. Thanks.

**Comment by balor** [comment]
+1

**Comment by DGideas** [comment]
According to the release note for 18.04:

> Upgrades from 16.04 LTS will not be enabled until a few days after the 18.04.1 release expected in late July.

https://wiki.ubuntu.com/BionicBeaver/ReleaseNotes

**Comment by OrangeDog** [comment]
@9527KKK then run `sudo apt autoremove` afterwards.

**Comment by 9527KKK** [comment]
"sudo apt autoremove" does not work at all!
etc.
By default U18 install GCC7.
"sudo apt autoremove" cannot remove GCC5 which is installed by U16 after upgrading from U16 to U18

**Comment by 9527KKK** [comment]
BTW, has WSL Ubuntu 18 LTS been released? If not, I think the issue status should be active instead of closed

**Comment by DarthSpock** [comment]
Funny, because I saw it was posted on Twitter 3 hours ago (vice your 2 hours ago comment) that they released a separate 18.04 appx in the Windows Store. So your desire to have a clean image now exists. Surprising they released it this soon but I'll take it once I can install to a non-system drive.

**Comment by DarthSpock** [comment]
It was there...Now the link is dead. Too many people trying to get it maybe?

**Comment by 9527KKK** [comment]
No idea. I don't know what the link is.

According https://docs.microsoft.com/en-us/windows/wsl/install-win10, the official way to install is through Windows Store.

I tried 3 hours ago. It was still U16.

**Comment by DarthSpock** [comment]
https://t.co/D5XtUFDzIR is the link provided from Twitter (don't need Twitter account to navigate link). But the page is dead for now. They also need to place it in the Linux on Windows collection.

**Comment by 9527KKK** [comment]
The link is working now. Thanks for your info.

**Comment by guijianchou** [comment]
nice, the link works. but not in " bash " search result so far on window store.

**Comment by 9527KKK** [comment]
Window Store search really sucks. That link is not searchable at all!

**Comment by benhillis** [comment]
It takes a while for apps to fully show up in the store, please be patient - it's coming.

**Comment by DarthSpock** [comment]
@benhillis did you get any information on when the stand-alone appx will be available?

**Comment by benhillis** [comment]
@DarthSpock - What do you mean by stand-alone appx. Server installs?

**Comment by DarthSpock** [comment]
Correct. This doc: https://docs.microsoft.com/en-us/windows/wsl/install-on-server which is desperately in need of updating with all the distros we have out now.

**Comment by benhillis** [comment]
@DarthSpock - Working on getting those documents updated.

**Comment by DarthSpock** [comment]
Well look what I finally see:

![image](https://user-images.githubusercontent.com/33032910/39931086-63e2df7c-550a-11e8-977a-1b41b5ef7bef.png)

**Comment by onomatopellan** [comment]
@DarthSpock Does it work for you? When I press the "Get the app" button it never downloads and refreshes the screen in a loop. This is on ver 1803.

I think they said on Build this would work on Windows on Arm so maybe is not 100% PC compatible yet.

**Comment by DarthSpock** [comment]
I got a Twitter notification in my email saying it was available for both amd64 and arm. I unfortunately can't download it atm due to corporate policy (I browsed the website vice Store app since the Store is disabled on my machine). I will test once I get off work

**Comment by DarthSpock** [comment]
They may be using the app as a test-platform for the cross-platform Store initiative. Also Win 10 S isn't supported if that's what you're running. Do you get an error code? If so, can you screen shot it and post it here?

**Comment by onomatopellan** [comment]
This is Windows 10 Pro build 17134.48. No error. Nothing in Event Viewer. Store app just refreshes when you press the button. It looks like a placeholder or maybe is only available on US market.

**Comment by gurpreetshanky** [comment]
@onomatopellan. Just installed. Even compiled gcc 8.1 on it. Working perfectly.

**Comment by onomatopellan** [comment]
@gurpreetshanky Thanks for confirming it. i just noticed it's a problem I have with Store and nothing will install anyway. Time to use the Troubleshoot I guess.

**Comment by 9527KKK** [comment]
Yesterday morning, it worked in on my first machine. But when I tried the link on another machine later, my download got stuck.

**Comment by onomatopellan** [comment]
Troubleshooter didn't help but I finally fixed my problem closing the user session inside the Store App and logging in again with the same user account.

**Comment by MichaelTong** [comment]
Is it possible that upgrade the current Ubuntu to Ubuntu 18.04?

**Comment by Biswa96** [comment]
See that at first of this discussion.

**Comment by aps120797** [comment]
Hey there all, here's and update.

**Ubuntu 18.04 is available on the Microsoft Store.**

[Ubuntu 18.04 on Microsoft Store](https://www.microsoft.com/store/productId/9N9TNGVNDL3Q)

You can simply download it from there.

Here's an official documentation by Microsoft on the same: https://blogs.msdn.microsoft.com/commandline/2018/05/15/build-2018-recap/

**Comment by sunilmut** [comment]
The 18.04 store links were broken in between for sometime, but, were fixed soon. It should work now.

**Comment by jacktose** [comment]
Ubuntu 18.04 on store: https://www.microsoft.com/store/productId/9N9TNGVNDL3Q

(I'm on Win 1803 (which should have been 1804) and it's working fine!)

**Comment by Biswa96** [comment]
Is this new in Store Ubuntu 16.04? https://www.microsoft.com/en-us/store/p/ubuntu-1604/9pjn388hp8c9

**Comment by DarthSpock** [comment]
Looks like Canonical is in the middle of replacing the Ubuntu appx to ensure users know which is 16.04 and 18.04. Gotta love user error. I was unable to search in the Store itself, required the link to navigate to it.

**Comment by DGideas** [comment]
@sunilmut It means we would see such as "ubuntu1804", "ubuntu2004" instead of legacy version of "ubuntu"(ubuntu1604) in the future?

**Comment by lewisdonofrio** [comment]
Been working for me and My Daily Driver for the past three months at 18.04 on w10 (corp locked down image but I spend all my days in XFCE and ubuntu 18.04 goodness - http://www.tinyurl.com/donofrioworkdesk)

**Comment by lewisdonofrio** [comment]
![1804-startup-console-messages-xfce4-working-seems-better-with-dbus-started](https://user-images.githubusercontent.com/3161117/41117860-327a6b28-6a5c-11e8-9737-64c5d20a467f.PNG)
Only had to do little dbus fix for etables, also ignoring all the console errors or else I'd go crazy lol....all and all it works awesomely for me...

**Comment by lewisdonofrio** [comment]
Spoke too soon - reloaded computer with current steps from www.tinyurl.com/donofrioworkdesk and I tripped over this when I was dist-upgrading from stock cloud ubuntu 18.04 to current code here is what I got:
***SNIPPED from console window***
To run a command as administrator (user "root"), use "sudo <command>".
See "man sudo_root" for details.

donofrio@LTAL0111:~$ sudo apt-get update && sudo apt-get upgrade -y && sudo apt-get dist-upgrade -y && sudo apt-get autoremove -y && sudo apt-get autoclean -y && sudo sync && sudo sync
[sudo] password for donofrio:
Get:1 http://security.ubuntu.com/ubuntu bionic-security InRelease [83.2 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic InRelease [242 kB]
Get:3 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [83.2 kB]
Get:4 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
Get:5 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [83.6 kB]
Get:6 http://security.ubuntu.com/ubuntu bionic-security/main Translation-en [30.3 kB]
Get:7 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [29.1 kB]
Get:8 http://security.ubuntu.com/ubuntu bionic-security/universe Translation-en [14.7 kB]
Get:9 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [1176 B]
Get:10 http://security.ubuntu.com/ubuntu bionic-security/multiverse Translation-en [632 B]
Get:11 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages [1019 kB]
Get:12 http://archive.ubuntu.com/ubuntu bionic/main Translation-en [516 kB]
Get:13 http://archive.ubuntu.com/ubuntu bionic/restricted amd64 Packages [9184 B]
Get:14 http://archive.ubuntu.com/ubuntu bionic/restricted Translation-en [3584 B]
Get:15 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages [8570 kB]
Get:16 http://archive.ubuntu.com/ubuntu bionic/universe Translation-en [4941 kB]
Get:17 http://archive.ubuntu.com/ubuntu bionic/multiverse amd64 Packages [151 kB]
Get:18 http://archive.ubuntu.com/ubuntu bionic/multiverse Translation-en [108 kB]
Get:19 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [123 kB]
Get:20 http://archive.ubuntu.com/ubuntu bionic-updates/main Translation-en [47.6 kB]
Get:21 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [80.3 kB]
Get:22 http://archive.ubuntu.com/ubuntu bionic-updates/universe Translation-en [36.5 kB]
Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [1736 B]
Get:24 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse Translation-en [1236 B]
Get:25 http://archive.ubuntu.com/ubuntu bionic-backports/universe amd64 Packages [2684 B]
Get:26 http://archive.ubuntu.com/ubuntu bionic-backports/universe Translation-en [1136 B]
Fetched 16.3 MB in 39s (413 kB/s)
Reading package lists... Done
W: http: aptMethod::Configuration: could not load seccomp policy: Invalid argument
W: http: aptMethod::Configuration: could not load seccomp policy: Invalid argument
W: store: aptMethod::Configuration: could not load seccomp policy: Invalid argument
Reading package lists... Done
Building dependency tree
Reading state information... Done
Calculating upgrade... Done
The following packages have been kept back:
  apport apt apt-utils bind9-host btrfs-progs cryptsetup-bin curl dirmngr dnsutils dpkg fonts-ubuntu-font-family-console gnupg gnupg-agent
  isc-dhcp-client libapparmor-perl libapt-pkg5.0 libbind9-160 libcurl3 libgnutls30 libidn2-0 libirs160 libisccc160 libisccfg160 liblwres160 liblxc1
  libnss-systemd libpam-systemd libperl5.26 libpsl5 libsystemd0 nplan perl perl-base python3-apport rsyslog strace systemd ubuntu-server
The following packages will be upgraded:
  apparmor at base-files bash bash-completion bsdutils btrfs-tools byobu ca-certificates cloud-guest-utils cloud-init command-not-found
  command-not-found-data console-setup console-setup-linux cryptsetup debconf debconf-i18n dh-python distro-info-data dmeventd dmsetup dnsmasq-base
  e2fslibs e2fsprogs eatmydata ebtables ethtool fdisk file friendly-recovery gcc-7-base gcc-8-base geoip-database gettext-base gir1.2-glib-2.0 git
  git-man gnupg-l10n gpgv hdparm initramfs-tools initramfs-tools-bin initramfs-tools-core iproute2 isc-dhcp-common iso-codes keyboard-configuration
  language-selector-common libapparmor1 libapt-inst2.0 libatm1 libblkid1 libc-bin libc6 libcom-err2 libcomerr2 libcurl3-gnutls libdb5.3
  libdevmapper-event1.02.1 libdevmapper1.02.1 libdrm-common libdrm2 libeatmydata1 libelf1 libext2fs2 libfdisk1 libfreetype6 libgcc1 libgcrypt20
  libgdbm5 libgirepository-1.0-1 libglib2.0-0 libglib2.0-data libgraphite2-3 libharfbuzz0b libicu60 libidn11 libkeyutils1 liblvm2app2.2 liblvm2cmd2.02
  liblz4-1 libmagic-mgc libmagic1 libmount1 libmpdec2 libncurses5 libncursesw5 libnghttp2-14 libnih1 libpam-modules libpam-modules-bin libpam-runtime
  libpam0g libplymouth4 libpolkit-agent-1-0 libpolkit-backend-1-0 libpolkit-gobject-1-0 libpopt0 libprocps6 libpython3-stdlib libpython3.6
  libpython3.6-minimal libpython3.6-stdlib libseccomp2 libselinux1 libsemanage-common libsemanage1 libslang2 libsmartcols1 libss2 libssl1.0.0
  libssl1.1 libstdc++6 libtinfo5 libudev1 libuuid1 libxcb1 locales lshw lvm2 lxcfs lxd lxd-client man-db mawk mdadm mlocate mount multiarch-support
  nano ncurses-base ncurses-bin ncurses-term netcat-openbsd open-iscsi open-vm-tools openssh-client openssh-server openssh-sftp-server openssl
  pastebinit patch perl-modules-5.26 plymouth plymouth-theme-ubuntu-text policykit-1 popularity-contest powermgmt-base procps publicsuffix
  python-apt-common python3 python3-apt python3-cffi-backend python3-commandnotfound python3-cryptography python3-debconf python3-distro-info
  python3-distupgrade python3-distutils python3-gdbm python3-lib2to3 python3-minimal python3-pkg-resources python3-problem-report python3-serial
  python3-software-properties python3-update-manager python3.6 python3.6-minimal rsync sensible-utils snapd software-properties-common sosreport
  squashfs-tools ssh-import-id systemd-sysv tcpdump tzdata ubuntu-advantage-tools ubuntu-keyring ubuntu-minimal ubuntu-release-upgrader-core
  ubuntu-standard ucf udev unattended-upgrades update-manager-core update-notifier-common util-linux uuid-runtime vim vim-common vim-runtime vim-tiny
  vlan wget xdg-user-dirs xfsprogs xxd
202 upgraded, 0 newly installed, 0 to remove and 38 not upgraded.
Need to get 90.0 MB of archives.
After this operation, 18.9 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 base-files amd64 10.1ubuntu2 [58.2 kB]
.
.
.
Get:199 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 cloud-init all 18.2-27-g6ef92c98-0ubuntu1~18.04.1 [356 kB]
Get:200 http://archive.ubuntu.com/ubuntu bionic/main amd64 open-vm-tools amd64 2:10.2.0-3ubuntu3 [539 kB]
Get:201 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 ssh-import-id all 5.7-0ubuntu1.1 [10.9 kB]
Get:202 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 vlan amd64 1.9-3.2ubuntu6 [30.7 kB]
Fetched 90.0 MB in 2min 25s (620 kB/s)
Extracting templates from packages: 100%
Preconfiguring packages ...
(Reading database ... 27025 files and directories currently installed.)
Preparing to unpack .../base-files_10.1ubuntu2_amd64.deb ...
Unpacking base-files (10.1ubuntu2) over (10ubuntu1) ...
Setting up base-files (10.1ubuntu2) ...
Installing new version of config file /etc/issue ...
Installing new version of config file /etc/issue.net ...
Installing new version of config file /etc/lsb-release ...
(Reading database ... 27026 files and directories currently installed.)
Preparing to unpack .../bash_4.4.18-2ubuntu1_amd64.deb ...
preinst: ../sysdeps/unix/sysv/linux/spawni.c:368: __spawnix: Assertion `ec >= 0' failed.
dpkg: error processing archive /var/cache/apt/archives/bash_4.4.18-2ubuntu1_amd64.deb (--unpack):
 new bash package pre-installation script subprocess was killed by signal (Aborted), core dumped
dmesg: read kernel buffer failed: Function not implemented
                                                          update-alternatives: using /usr/share/man/man7/bash-builtins.7.gz to provide /usr/share/man/man7/builtins.7.gz (builtins.7.gz) in auto mode
Errors were encountered while processing:
W: http: aptMethod::Configuration: could not load seccomp policy: Invalid argument
E: Sub-process /usr/bin/dpkg returned an error code (1)
donofrio@LTAL0111:~$
**SNIPPED**
now I'm stuck in unpatched session, I'm open to suggestions.

**Comment by jacktose** [comment]
@lewisdonofrio 18.04 is available. This issue is closed. I think you want a new issue.

**Comment by lewisdonofrio** [comment]
I know that I'm doing this "without the store" so I'm using a cloud image then I update it to current patch level.  I've gotten around etables issues with dbus fix but now I'm stick at the issue ;(

**Comment by onomatopellan** [comment]
@lewisdonofrio From your screenshot I can see you don't have installed latest April 2018 update. You will need to install it to fix the '__spawnix: Assertion `ec >= 0' failed.' bug.

**Comment by DarthSpock** [comment]
Please stop spamming this issue thread. This was about the official image being available. It is and as iterated several times above, just awaiting the server install option. Please take any other issues regarding 18.04, from the official image or from an unofficial to a new issue. It is expected for there to be issues regarding 18.04 that could stem from either the distro itself or WSL but this issue is NOT the landing zone for either.

**Comment by lewisdonofrio** [comment]
I worked around this issue by chance, installing inxi at the very start of my process, I do sudo apt-get update then sudo apt-get install inxi -y and then full sudo apt-get distr-upgrade and poof it works, so while that segfault still exists (this problem will return in some form guessing) I was able to proceed on to full xfce4/gui victory for 18.04 tls.
(my without the store load process is at http://www.tinyurl.com/donofrioworkdesk - now with raw text steps along with ppt sides and the one zip file that you place it in download dirs and it will make rootfs there.)

**Comment by SRGOM** [comment]
@mikegitb you always could using the excellent tool from @rolisoft

**Comment by MikeGitb** [comment]
@SRGOM: What exactly are you referring to? Installing Ubuntu 16.04 side by side with 18.04 is already possible.

**Comment by thedonquixotic** [comment]
Hey newbie developer here trying to automate my fresh install process etc and I was curious if there was an updated way to powershell script fresh install WSL Ubuntu 18.04? I haven't seen a link to install it like with 16.04 so I thought maybe I could use get-appx add-appx to install it like I have with other packages found in the MS store, but it has not worked for me so far. Any suggestions?

I know there are also people hosting third party versions on cloud platforms etc, but I'm hoping there is an official MS way of doing it so I don't have to change my script in 3 weeks or something. I'm hoping to share it with friends and I'd like to do something as anti-fragile and long term solution as possible.

**Comment by lewisdonofrio** [comment]
Have you seen my steps, scripts?  I have all the details about my WSL 18.04 TLS (keep in context I do not have "The Store" so this is not using the appx it is for our corp w10 image.) www.tinyurl.com/donofrioworkdesk
![xfce4-up-capture](https://user-images.githubusercontent.com/3161117/42547165-c63ba25e-848e-11e8-99d9-f2d0b4bd65c1.PNG)

**Comment by thedonquixotic** [comment]
I did see it but wasn't sure how long your files would be up. Also I'm kind
of a newbie to coding (my day job is ux design) but are you running your
Linux graphical shell inside of windows? How'd you do that? Is it a vm or
is that actually running off of wsl?

**Comment by lewisdonofrio** [comment]
My files will always be up, I may add more scripts .sh to do more things that are needed, like the fix-dbus and the like.  keep on context I use the xfce4 gui daily and if you don't need the gui the stop when my steps say "stop go not further"

**Comment by thedonquixotic** [comment]
Sweet thanks. I'll try it out 👍

**Comment by lewisdonofrio** [comment]
Still working on getting .cache created and owned by your usernam.  Oh and I suggest using a different passwd on the LTS install than your w10 login, aka same username is fine just change the passwd when it asks you for a password make it different (I don't trust things lol) is all.

**Comment by lewisdonofrio** [comment]
fwiw, I lurk always on freenode (I use hexchat daily from within xfce4 wsl setup) in the #ubuntu-on-windows

**Comment by DarthSpock** [comment]
I could of swore I commented to answer the question @thedonquixotic . Must of forget to press the comment button before closing the page. Please keep GUI discussion in the appropriate thread please. As far as installing 18.04 goes, there are several ways to do it. If you're looking for the offline server install, we're still awaiting word on it. You would use PowerShell for this method. Since there is no offline appx to install, I recommend reading this blog regarding Ubuntu: https://blogs.msdn.microsoft.com/commandline/2018/07/09/upgrading-ubuntu/

**Comment by lewisdonofrio** [comment]
My steps are for offline 18.04 TLS (well at least "without store" how to do it if your Corp Work build has no store installed.) for my setup you still need the internet or at least rsync'ed local copy of the repo files.  Keep in mind I'm just trying to help out this great 'community.'

**Comment by thedonquixotic** [comment]
👍

**Comment by lewisdonofrio** [comment]
@thedonquixotic How did your install attempt go?  I'm on irc/freenode/#ubuntu-on-windows for chat (to keep small talk off this thread)  feel free to email me if you have issue/thoughts.

**Comment by thedonquixotic** [comment]
I got it working. Thanks Lewis!

Sincerely,David Aslan French  
Jackalope.tech Design X Code: Pushing the boundaries of design, art, and code.
 jackalope.tech  
Find me on Facebook  Let's connect!  








On Fri, Jul 13, 2018 7:09 AM, Lewis L. Donofrio notifications@github.com  wrote:
@thedonquixotic  How did your install attempt go? I'm on
irc/freenode/#ubuntu-on-windows for chat (to keep small talk off this thread)
feel free to email me if you have issue/thoughts.
